externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/result/ResultSet.java

```diff
@@ -53,11 +53,6 @@ private ResultSet(
     this.resultKind = Preconditions.checkNotNull(resultKind, "resultKind must not be null");
     this.columns = Preconditions.checkNotNull(columns, "columns must not be null");
     this.data = Preconditions.checkNotNull(data, "data must not be null");
-    if (!data.hasNext()) {
-      Preconditions.checkArgument(
-          columns.size() == data.next().getArity(),
```

Comment:
good catch, if we want to check, we would expect the type of data is a `List`.

Then we can do:

```
if (!data.isEmpty()) {
  Preconditions.checkArgument(columns.size() == data.get(0).getArity()
}
```

But for Iterator, we can ignore the check here.